### PR TITLE
Log actual checkout hash instead of generic refspec

### PR DESCRIPTION
**Pull request by mark-vieira**
This PR fixes our logging such that we report the actual checkout hash for snapshot BWC distributions instead of the refspec. This is important for builds that do *not* use `bwc.checkout.align` since the reported hash looks like:

```
Checkout hash for :distribution:bwc:bugfix is elastic/7.3
```

This isn't very helpful for debugging or reproduction.

### CODE REVIEW

File: distribution/bwc/build.gradle [hunk 1]
```diff
@@ -105,7 +105,7 @@ bwcVersions.forPreviousUnreleased { BwcVersions.UnreleasedVersionInfo unreleased
         dependsOn fetchLatest
         doLast {
             String refspec = System.getProperty("bwc.refspec.${bwcBranch}") ?: System.getProperty("tests.bwc.refspec.${bwcBranch}") ?: "${remote}/${bwcBranch}"
-            if (System.getProperty("bwc.checkout.align") != null || System.getProperty("tests.bwc.checkout.align") != null) {
```
Comment (rjernst) [inline distribution/bwc/build.gradle]: Is this an unrelated change? I'm curious why we have the tests.bwc.refspec above here as well.

**Review by rjernst — APPROVED** [review]
LGTM

File: distribution/bwc/build.gradle (hunk 1 above)
Comment (mark-vieira) [reply distribution/bwc/build.gradle]: This was for backwards compatibility with CI when we renamed this property. Just now remembering to go back and remove this. See https://github.com/elastic/infra/pull/13812.

**Review by alpar-t — APPROVED** [review]
LGTM

**Comment by mark-vieira** [comment]
@elasticmachine update branch

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by mark-vieira** [comment]
@elasticmachine update branch

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/2
